### PR TITLE
meta(engine): Refine meta adjust meta pkg (Part2)

### LINKED ISSUE #6180: (meta)engine: adjust meta pkg directory
### Is your feature request related to a problem?

No

### Describe the feature you'd like

adjust meta pkg directory

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve?

Issue Number: close #6180 

### What is changed and how it works?
1. move all inner implement to internal
2. Change 'metaclient' dir to 'common' dir

Principle of the adjustment:
1. Put `client.go` and `example.go` in first layer of meta for easier usage
2. `model` dir is the common stucture and interface definition of `kvclient`. (an independent dir for no circle import)
3. `internal` dir is the inner implement for kvclient, like `etcdkv` and `sqlkv`
4. `mock` dir is for mock test.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: engine/pkg/meta/common/client.go [hunk 1]
```diff
@@ -11,27 +11,45 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package extension
+package common
```
Comment (lonng) [inline engine/pkg/meta/common/client.go:14]: I think `common` is not a meaningful package name. Can we give a more descriptive name?

**Review by lonng — COMMENTED** [review]
rest LGTM

File: engine/pkg/meta/common/client.go [hunk 2]
```diff
@@ -11,27 +11,45 @@
...
 
 import (
 	"context"
-
-	"github.com/pingcap/tiflow/engine/pkg/meta/metaclient"
 )
 
+// Client defines some basice method used as a meta client
```
Comment (amyangfei) [inline engine/pkg/meta/common/client.go:20]:
```suggestion
// Client defines some basic method used as a meta client
```

File: engine/pkg/meta/common/client.go (hunk 1 above)
Comment (maxshuang) [reply engine/pkg/meta/common/client.go:14]: It's for the placement of common inteface and structure for meta, although this name is too `common`.
I will try.

File: scripts/generate-engine-mock.sh [hunk 3]
```diff
@@ -25,8 +25,8 @@ fi
 
 echo "generate dataflow engine mock code..."
 
-"$MOCKGEN" -package mock github.com/pingcap/tiflow/engine/pkg/meta/metaclient KVClient \
-	>engine/pkg/meta/kvclient/mock/client_mock.go
+"$MOCKGEN" -package mock github.com/pingcap/tiflow/engine/pkg/meta/common KVClient \
```
Comment (CharlesCheung96) [inline scripts/generate-engine-mock.sh:28]: Whether the script should be placed inside the package? e.g:https://github.com/pingcap/tiflow/blob/master/cdc/redo/writer/writer.go#L41

File: engine/pkg/meta/common/client.go (hunk 1 above)
Comment (maxshuang) [reply engine/pkg/meta/common/client.go]: Have Changed to 'model'.

File: scripts/generate-engine-mock.sh (hunk 3 above)
Comment (maxshuang) [reply scripts/generate-engine-mock.sh:28]: Seems that `mockery` and `gomock` are tools for easy mocking. 
May be we can do it in other pr if we really need it.

File: engine/executor/server.go [hunk 4]
```diff
@@ -44,7 +44,7 @@ import (
 	"github.com/pingcap/tiflow/engine/pkg/deps"
 	"github.com/pingcap/tiflow/engine/pkg/externalresource/broker"
 	"github.com/pingcap/tiflow/engine/pkg/externalresource/storagecfg"
-	extkv "github.com/pingcap/tiflow/engine/pkg/meta/extension"
+	metaclient "github.com/pingcap/tiflow/engine/pkg/meta/model"
```
Comment (gozssky) [inline engine/executor/server.go:47]: `metaclient` -> `metamodel` ? Ditto for all places.

File: engine/pkg/meta/model/client.go [hunk 5]
```diff
@@ -11,27 +11,45 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package extension
+package common
```
Comment (amyangfei) [inline engine/pkg/meta/model/client.go:14]: Why is it still `package common`?

File: engine/executor/server/mocks/metastore_mock.go [hunk 6]
```diff
@@ -9,8 +9,7 @@ import (
 	reflect "reflect"
 
 	gomock "github.com/golang/mock/gomock"
-	extension "github.com/pingcap/tiflow/engine/pkg/meta/extension"
-	metaclient "github.com/pingcap/tiflow/engine/pkg/meta/metaclient"
+	common "github.com/pingcap/tiflow/engine/pkg/meta/model"
```
Comment (gozssky) [inline engine/executor/server/mocks/metastore_mock.go:12]: Please unify all the import alias.

**Review by CharlesCheung96 — APPROVED** [review]

File: engine/pkg/meta/model/client.go (hunk 5 above)
Comment (maxshuang) [reply engine/pkg/meta/model/client.go]: Fix it. 
The `make engine` and `make check` can't check this situation.

File: engine/executor/server.go (hunk 4 above)
Comment (maxshuang) [reply engine/executor/server.go]: Has changed to `metaModel`.

File: engine/executor/server/mocks/metastore_mock.go (hunk 6 above)
Comment (maxshuang) [reply engine/executor/server/mocks/metastore_mock.go]: This file is autogenerated by `mockgen`, so the alias here may be different  from other cases.

Comment (maxshuang) [reply engine/executor/server/mocks/metastore_mock.go]: Fix it as `model "github.com/pingcap/tiflow/engine/pkg/meta/model"`

**Review by amyangfei — APPROVED** [review]

**Review by gozssky — APPROVED** [review]

**Comment by maxshuang** [comment]
@amyangfei  
Please tag a /can-merge flag for this pr if you are available. Thanks.

**Pull request merged**